### ISSUE TITLE
How to deal with a multi label classification of sparse vector

### ISSUE THREAD

**Issue by 12gjang** [fchollet/keras#2578]
My data looks like these,

X_t=0 = [0,0,0,1,0,0,1,0, .....]
X_t=1 = [0,0,0,0,0,0,0,0, .....]
X_t=2 = [1,0,0,0,0,0,0,0, .....]
.......

This is a series of  sparse vectors

And I want to train my model to predict the event of next time step.

I am using LSTM with binary cross entropy + adam 

But accuracy is not good. (Always stuck in 0.75 acc, it is similar to percentage of 0 in my data )

I think It is always converged to all 0 result.

Why I think like this, when I use init='zero' , It starts with 0.75 accuracy

That is the my convergence point, when I didn't use init='zero'

There are many zeros in my training data.

But I can not use 'softmax', because multiple choice  is possible...

The distance between the all zero vector and target is very close.

I am worrying about case that my model just predicts all zero vector in many cases.

Is this issue not worth discussing ?

**Comment by algoseer** [comment]
If you train using sigmoid and binary crossentropy only, your classification is going to be biased to the majority class which it seems is 0 for all the classes in your case. I would suggest to use class_weights to deal with this issue or better still write your own loss fucntion.
